### PR TITLE
Simplifications to Avocado's executables (aka scripts)

### PR DESCRIPTION
**Pull request by clebergnu**
A couple of cleanups/simplifications for the avocado executables (aka scripts).

### CODE REVIEW

File: scripts/avocado [hunk 1]
```diff
@@ -56,10 +56,6 @@ def handle_exception(*exc_info):
     sys.exit(-1)
 
 
-if __name__ == '__main__':
-    sys.excepthook = handle_exception
```
Comment (ldoktor) [inline scripts/avocado]: This check was on purpose moved here to provide nicer message on `AvocadoApp` import crash, so please don't remove it...

File: scripts/avocado [hunk 2]
```diff
@@ -56,14 +56,6 @@ def handle_exception(*exc_info):
     sys.exit(-1)
 
 
-# simple magic for using scripts within a source tree
-basedir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
-if os.path.isdir(os.path.join(basedir, 'avocado')):
-    os.environ['PATH'] += ":" + os.path.join(basedir, 'scripts')
-    os.environ['PATH'] += ":" + os.path.join(basedir, 'libexec')
-    sys.path.append(basedir)
```
Comment (ldoktor) [inline scripts/avocado]: I'll have to give it a try (after the LTS) but if that works well I'm definitely for this.

File: scripts/avocado (hunk 1 above)
Comment (clebergnu) [inline scripts/avocado]: Then the avocado module import can be moved as well (it doesn't affect style rules/checks).

Comment (ldoktor) [inline scripts/avocado]: The point is that when avocado import fails we do want to store the info about this failure. If we moved the import before setting the excepthook we won't get the treatment on import error.

**Comment by clebergnu** [comment]
Closing in favor of #2171

**Pull request closed without merging**